source/structuring.rst

```diff
@@ -14,402 +14,468 @@ functions is better than copying-and-pasting duplicate bits of code
 everywhere they are used.  Likewise in JSON Schema, for anything but
 the most trivial schema, it's really useful to structure the schema
 into parts that can be reused in a number of places.  This chapter
-will present some practical examples that use the tools available for
-reusing and structuring schemas.
+will present the tools available for reusing and structuring schemas
+as well as some practical examples that use those tools.
 
-Reuse
------
+.. index::
+   single: schema identification
+   single: structuring; schema identification
 
-For this example, let's say we want to define a customer record, where
-each customer may have both a shipping and a billing address.
-Addresses are always the same---they have a street address, city and
-state---so we don't want to duplicate that part of the schema
-everywhere we want to store an address.  Not only would that make the
-schema more verbose, but it makes updating it in the future more
-difficult.  If our imaginary company were to start doing international
-business in the future and we wanted to add a country field to all the
-addresses, it would be better to do this in a single place rather than
-everywhere that addresses are used.
+.. _schema-identification:
+
+Schema Identification
+---------------------
+
+Like any other code, schemas are easier to maintain if they can be
+broken down into logical units that reference each other as necessary.
+In order to reference a schema, we need a way to identify a schema.
+Schema documents are identified by URIs. Schemas documents are not
+required to have an identifier, but obviously you will need one if you
+want to reference the schema from another. In the following sections
+we will see how the "identifier" for a schema is determined.
 
-So let's start with the schema that defines an address::
+.. index::
+   single: retrieval URI
+   single: structuring; schema identification; retrieval URI
+
+.. _schema-identification:
+
+Retrieval URI
+~~~~~~~~~~~~~
+
+Even though schemas are identified by URIs, those identifiers are not
+necessarily network accessible. They are just identifiers.
+Implementations generally provide a way for you to load schemas into
+an internal schema database and they retrieve schemas from there when
+they are referenced.
+
+How ever the schema is retrieved, the URI used to fetch the schema is
+known as the "retrieval URI" and is always an absolute URI. It's often
+possible to pass a literal schema to an implementation in which case
```

Comment:
I like the rewrite very much. It's very comprehensible for me. There's just the term *literal schema* which isn't as obvious to me. I'm not a native speaker so can't tell how common *literal* is being used. So far I guess in this context it means to pass an already loaded in-memory object or a serialized form of a schema to a schema processor which hasn't been resolved by the schema processor itself.

I do not propose using a different term. But it's used at least once more in this section and maybe it could be defined or explained in an additional sentence in some way, particularly if it is some kind of JSON-Schema terminology.

If you don't mind I'll do a little review....

Reply:
Thank you! This is really helpful feedback.

The word "literal" is a term used in programming languages (https://en.wikipedia.org/wiki/Literal_(computer_programming)). However, I used it wrong. That line should say "schema literal", not "literal schema". You might be familiar with the term "string literal", meaning an expression that represents a string as opposed to a variable that points to a string. I was attempting to convey the same kind of idea, an expression representing a schema, but with no variable giving it a "name" you can refer to it by. It can only be used in the expression it is part of because there is way to refer to it beyond that expression.

I think if I had used the term correctly it would have been more clear but, It's still probably too much jargon for this guide. We want this to be as accessible as possible. I'll see what I can do to make it more clear.

Reply:
I decided to change the terminology from "schema literal" to "anonymous schema" and I added a sentence defining what that means. Hopefully, that makes things a little more clear.

Reply:
Thank you for your openess. You're right. I know string literal or the "object literal notation" (`{ prop: "value" }`) in JavaScript but *anonymous schema* sounds much better to me. Without any context I could immediately guess that it is probably a schema without an `$id` identifier.